tw-tasks-core/src/main/java/com/transferwise/tasks/helpers/kafka/AdminClientTopicPartitionsManager.java

```diff
@@ -31,7 +33,9 @@
   @Override
   public void setPartitionsCount(String topic, int partitionsCount) {
     ExceptionUtils.doUnchecked(() -> {
-      AdminClient adminClient = AdminClient.create(kafkaConfiguration.getKafkaProperties().buildAdminProperties());
+      Map<String, Object> adminConfig = kafkaConfiguration.getKafkaProperties().buildAdminProperties();
+      adminConfig.put(AdminClientConfig.CLIENT_ID_CONFIG, "");
+      AdminClient adminClient = AdminClient.create(adminConfig);
```

Comment:
admin client implement auto closable. If you need to use it only in this method, you'd need to close the admin client when job is done

Reply:
yep, we do close the client as well. But this code can also be called in parallel.

Reply:
called in parallel should be safe as if we pass empty they use atomicInteger so should be safe.